code/codenames.py

```diff
@@ -1,24 +1,36 @@
 import numpy as np
 import json
 from termcolor import colored, cprint
-from player import Player
+from player import Player, AI, Human
 from wordBase import WordBase, Word
+from data_encoding import data_encodings
 import time
 import os
 
 
 class Codenames:
     def __init__(self,
-                team_a_ms, team_a_gs, team_b_ms, team_b_gs,
-                data_file='../data/processed_data/cosine_sim_mat.npz',
+                team_a_ms, team_a_gs, team_b_ms, team_b_gs, alg=None,
+                data_file=None,
                 seed=None
     ):
         if seed != None:
             self.seed = seed
         else:
             self.seed = np.random.randint(2**32 - 1)
         np.random.seed(self.seed)
-        self.word_base = WordBase(data_file)
+
+        if alg != None:
+            self.alg = alg
+        else:
+            self.alg = 1 ####replace with the best algorithm later
+        
+        if data_file != None:
+            self.data_file = data_encodings[data_file]
+        else:
+            self.data_file = '../../AI_dataset/cosine_sim_mat.npz' ####replace with best data set later
```

Comment:
ya because we can't have the data in the repo (not supposed to from lecture and files are too big). So all the data are in the AI_dataset which is outside the git repo. lmk if we want to do it another way.